Samples/BackgroundBlur/BackgroundBlur/common/critsec.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+namespace MediaFoundationSamples
+{
+
+
+    //////////////////////////////////////////////////////////////////////////
+    //  CritSec
+    //  Description: Wraps a critical section.
+    //////////////////////////////////////////////////////////////////////////
+
+    class CritSec
+    {
+    private:
+        CRITICAL_SECTION m_criticalSection;
+    public:
+        CritSec()
+        {
+            InitializeCriticalSection(&m_criticalSection);
```

Comment:
Using InitializeCriticalSection and the interlocked methods from Windows is fine, but in general I'd use the methods from the STL (mutex/unique_lock and atomic in this case). They come with C++ RAII, are cross-platform and will call the same or similar Windows APIs under the hood. But no need to refactor all the code now.